packages/xstate-lit/src/UseMachine.ts

```diff
@@ -0,0 +1,85 @@
+import { ReactiveController, ReactiveControllerHost } from 'lit';
+import {
+  Actor,
+  AnyStateMachine,
+  ActorOptions,
+  createActor,
+  EventFrom,
+  SnapshotFrom,
+  Subscription
+} from 'xstate';
+
+export class UseMachine<TMachine extends AnyStateMachine>
```

Comment:
Since this is a controller and not a React hook, I'd recommend against the "Use" name, and call this something like `MachineController`

Reply:
As someone more planted in the Lit ecosystem, I think it'd be nice to match Lit naming too - which just follows the somewhat cross-language standard OO noun-phrase vs verb-phrase naming. The class should be the noun-phrase name of the category of thing that the instances are, while functions should be verb-phrases.

`UseMachine` is a verb-phrase, and I think it reads a bit weird to have statements like:

```ts
this.fetchController = new UseMachine(...);
```

Because this is saying that `fetchController` _is_ a "UseMachine". But what is a "UseMachine"? It's a controller - an object with a verb-phrased name. But it sounds like a function.

This is why most reactive controllers are named {X}Controller.

It's possible to vend a function that makes the class for you, so maybe that's better:

```ts
this.fetchController = useMachine(...);
```

But even then, in the Lit ecosystem I'd still want to shy away from the "use" name because people tend to think that it's a hook and has React-like "rule of hooks" semantics, when it doesn't.